### PR TITLE
Reward Campaigns: Added emergency cancel campaign and add/remove set reward origin extrinsics + multi-currency campaigns support

### PR DESCRIPTION
**Pull request by chexware**

### CODE REVIEW

File: pallets/reward/src/lib.rs [hunk 1]
```diff
@@ -292,12 +334,17 @@ pub mod pallet {
...
 					Error::<T>::CampaignExpired
 				);
-				ensure!(amount <= campaign.cap, Error::<T>::RewardExceedCap);
 
-				campaign.cap -= amount;
-				Self::reward_put(campaign.trie_index, &to, &amount, &[]);
+				match campaign.cap {
+					RewardType::FungibleTokens(c, b) => {
+						ensure!(amount <= b, Error::<T>::RewardExceedCap);
+						campaign.cap = RewardType::FungibleTokens(c, b - amount);
```
Comment (justinphamnz) [inline pallets/reward/src/lib.rs:341]: This needs to be checked sub or saturating sub to ensure no overhead

File: pallets/reward/src/lib.rs [hunk 2]
```diff
@@ -318,11 +365,75 @@ pub mod pallet {
 			);
 
 			let fund_account = Self::fund_account_id(id);
-			let unclaimed_balance = campaign.reward - campaign.claimed;
-			T::Currency::transfer(&fund_account, &who, unclaimed_balance, AllowDeath)?;
-			Campaigns::<T>::remove(id);
+			match campaign.reward {
+				RewardType::FungibleTokens(_, r) => match campaign.claimed {
+					RewardType::FungibleTokens(c, b) => {
+						let unclaimed_balance = r - b;
```
Comment (justinphamnz) [inline pallets/reward/src/lib.rs:371]: All calculation please use saturating

**Review by justinphamnz — CHANGES_REQUESTED** [review]
Please ensure safe check

**Comment by chexware** [comment]
@justinphamnz  I've updated the subtraction to saturting_sub

**Review by justinphamnz — APPROVED** [review]
Thanks @chexware. Great work.

**Pull request merged**